internal/cow/cow.go

```diff
@@ -0,0 +1,70 @@
+package cow
+
+import (
+	"io"
+
+	"github.com/Microsoft/hcsshim/internal/schema1"
+)
+
+// Process is the interface for an OS process running in a container or utility VM.
+type Process interface {
+	// Close releases resources associated with the process and closes the
+	// writer and readers returned by Stdio. Depending on the implementation,
```

Comment:
Oh very interesting. Since its not `Stdio() (io.WriteCloser, etc)` how does this impl know when its safe to close them?

Reply:
It closes them when you call Close() on the process, which you would normally only do after Wait() returned.

Reply:
(This is not new--this got changed when I did the stdio refactor a while ago.)

Reply:
So the pattern for a caller is.
```go
c := Create()
in, out, serr := c.Stdio()
c.Wait()
// (wait for IO routines to finish)
c.Close()
```

Reply:
Right. Although this will be encapsulated into the new Cmd type in the next PR.

Reply:
Ok cool. That's what I remember you talking about. Will be nice to not duplicate the relay code.